### ISSUE TITLE
Box shadow animated

### ISSUE THREAD

**Issue by VictorPullzz** [software-mansion/react-native-reanimated#1296]
## Description
Hi guys! I was trying to make the box shadow appear smoothly when the user scrolls the block. But something doesn't work, tell me what is the problem?)

## Code
```
....
const [boxShadow, setBoxShadow] = useState<boolean>(false);

const scrollHandler = useAnimatedScrollHandler((event) => {
    console.log(event.contentOffset.y)
    setBoxShadow(event.contentOffset.y > 10);
});

const styleTopBox = useAnimatedStyle(() => {
	if (boxShadow) {
		return {
			...Platform.select({
				ios: {
					shadowColor: theme.colors.colorBoxShadow,
					shadowOffset: {
						width: 0,
						height: 3,
					},
					shadowOpacity: 0.27,
					shadowRadius: 4.65,
				},
				android: {
					elevation: 10,
				}
			})
		}
	}
        return {};
});
<Animated.View
	style={{
           backgroundColor:  '#000'
             ...styleTopBox
	}}
>
...
</Animated.View>
```
Before scroll:
![image](https://user-images.githubusercontent.com/23406402/94584569-7b746c00-0287-11eb-9cd0-f6a524c21a3a.png)

After scroll:
![image](https://user-images.githubusercontent.com/23406402/94584609-86c79780-0287-11eb-827b-d8c112fd6a0e.png)

### Package versions
- React: 16.13.1
- React Native: https://github.com/expo/react-native/archive/sdk-39.0.2.tar.gz
- React Native Reanimated: ^2.0.0-alpha.6

**Comment by terrysahaidak** [comment]
You should not spread the animated styles but pass it as a style

```tsx
<Animated.View style={[styleTopBox, { backgroundColor: '#000' }]></Animated.View>
```

**Comment by VictorPullzz** [comment]
@terrysahaidak I will check :)

**Comment by VictorPullzz** [comment]
@terrysahaidak Dont work for me :( 
![image](https://user-images.githubusercontent.com/23406402/94707269-ea1afd80-034b-11eb-88c2-879d5d01af76.png)

**Comment by jakub-gonet** [comment]
@VictorPullzz, please don't post screenshots of code, just use code-blocks.

**Comment by VictorPullzz** [comment]
> @VictorPullzz, please don't post screenshots of code, just use code-blocks.

Ah okay, I change

**Comment by VictorPullzz** [comment]
@jakub-gonet Do you know how to get animation to work for box-shadow? :)

**Comment by terrysahaidak** [comment]
Try this:

```tsx
const showBoxShadow = useSharedValue(false);

const scrollHandler = useAnimatedScrollHandler((event) => {
	showBoxShadow.value = event.contentOffset.y > 10;
});
const shadowColorStyle = useAnimatedStyle(() => {
	if (showBoxShadow.value) {
		return {
			shadowColor: '#000',
			shadowOffset: { width: 0, height: 2, },
			shadowOpacity: 0.27,
			shadowRadius: 2.65,
			elevation: 5,
		}
	} else {
		return {
			shadowColor: 'transparent',
			shadowOffset: { width: 0, height: 0, },
			shadowOpacity: 0,
			shadowRadius: 0,
			elevation: 0,
		}
	};
});

```

**Comment by VictorPullzz** [comment]
@terrysahaidak unfortunately dont work on ios, on android give this error: 
![image](https://user-images.githubusercontent.com/23406402/94734054-4fcdb080-0371-11eb-995f-a08a1a658643.png)

**Comment by terrysahaidak** [comment]
Could you create a [snack](snack.expo.io) then?

**Comment by karol-bisztyga** [comment]
You can use either state or shared value, here it does not really matter as `useAnimatedStyle` with no deps passed is going to be recreated on rerender with a proper value.

[This change](https://github.com/software-mansion/react-native-reanimated/pull/1304) fixes this problem on iOS but I was not able to reproduce this on Android(shadow box would not work without external lib). Let me know if the problem persists on Android(although shadow box was not visible I wasn't getting such errors whatsoever).

**Comment by VictorPullzz** [comment]
@terrysahaidak  Yep

**Comment by karol-bisztyga** [comment]
This has been closed automatically with iOS solution being merged.
@VictorPullzz if the problem persists, please reopen.

**Comment by wayne-tt** [comment]
Do not use object value to shadowOffset.

**Comment by charles-goode** [comment]
I was getting `TypeError: Attempting to define property on object that is not extensible` when merging styles like `style={[{/*other styles */ shadowOffset: {width: 0, height: 0 }}, animateStyle]}` where the animatedStyle was animating shadowRadius. It was fixed by moving the static definition of shadowOffset into the animatedStyle. Pretty weird and the error message wasn't helpful. I got it through process of elimination.
